front-end/src/components/Signup/Web3Signup.tsx

```diff
@@ -0,0 +1,298 @@
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { DeriveAccountInfo } from '@polkadot/api-derive/types';
+import { web3Accounts, web3Enable,web3FromSource } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
+import { stringToHex } from '@polkadot/util';
+import styled from '@xstyled/styled-components';
+import React, { useContext, useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Link } from 'react-router-dom';
+import { Divider, DropdownProps } from 'semantic-ui-react';
+import { ApiContext } from 'src/context/ApiContext';
+
+import ExtensionNotDetected from '../../components/ExtensionNotDetected';
+import { ModalContext } from '../../context/ModalContext';
+import { UserDetailsContext } from '../../context/UserDetailsContext';
+import { useAddressSignupConfirmMutation, useAddressSignupStartMutation } from '../../generated/graphql';
+import { useRouter } from '../../hooks';
+import { handleTokenChange } from '../../services/auth.service';
+import AccountSelectionForm from '../../ui-components/AccountSelectionForm';
+import Button from '../../ui-components/Button';
+import FilteredError from '../../ui-components/FilteredError';
+import { Form } from '../../ui-components/Form';
+import Loader from '../../ui-components/Loader';
+import getEncodedAddress from '../../util/getEncodedAddress';
+
+interface Props {
+	className?: string
+	toggleWeb2Signup: () => void
+}
+
+const APPNAME = process.env.REACT_APP_APPNAME || 'polkassembly';
+const NETWORK = process.env.REACT_APP_NETWORK || 'kusama';
+
+const SignupForm = ({ className, toggleWeb2Signup }:Props): JSX.Element => {
+	const [error, setErr] = useState<Error | null>(null);
+	const [address, setAddress] = useState<string>('');
+	const [email, setEmail] = useState<string>('');
+	const [username, setUsername] = useState<string>('');
+	const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
+	const [isAccountLoading, setIsAccountLoading] = useState(true);
+	const [extensionNotFound, setExtensionNotFound] = useState(false);
+	const [accountsNotFound, setAccountsNotFound] = useState(false);
+	const { history } = useRouter();
+	const { api, apiReady } = useContext(ApiContext);
+	const [addressSignupStartMutation] = useAddressSignupStartMutation();
+	const [addressSignupConfirmMutation, { loading }] = useAddressSignupConfirmMutation();
+	const currentUser = useContext(UserDetailsContext);
+	const { errors, handleSubmit, register } = useForm();
+	const { setModal } = useContext(ModalContext);
+
+	useEffect(() => {
+		if (!accounts.length) {
+			getAccounts();
+		}
+	}, [accounts.length]);
+
+	useEffect(() => {
+		if (!api) {
+			return;
+		}
+
+		if (!apiReady) {
+			return;
+		}
+
+		let unsubscribe: () => void;
+
+		api.derive.accounts.info(address, (info: DeriveAccountInfo) => {
```

Comment:
Let's remove all this logic, this is transparent to our users and shouldn't happen.

Reply:
That's what I meant to say in https://github.com/paritytech/polkassembly/pull/724#pullrequestreview-405169798

Reply:
removed